modules/drivers/athena/test/metabase/driver/athena_test.clj

```diff
@@ -12,31 +13,43 @@
+  [{:name "key", :database-type "int"}
+   {:name "data", :database-type "struct<name:string>"}])
 
 (def ^:private flat-schema-columns
-  [{:column_name "id", :type_name  "string"}
-   {:column_name "ts", :type_name "string"}])
+  [{:name "id", :database-type  "string"}
+   {:name "ts", :database-type "string"}])
 
-(deftest sync-test
+(deftest ^:parallel sync-test
   (testing "sync with nested fields"
-    (with-redefs [athena/run-query (constantly nested-schema)]
-      (is (= #{{:name              "key"
-                :base-type         :type/Integer
-                :database-type     "int"
-                :database-position 0}
-               {:name              "data"
-                :base-type         :type/Dictionary
-                :database-type     "struct"
-                :nested-fields     #{{:name "name", :base-type :type/Text, :database-type "string", :database-position 1}},
-                :database-position 1}}
-             (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))
+    (is (= #{{:name              "key"
+              :base-type         :type/Integer
+              :database-type     "int"
+              :database-position 0}
+             {:name              "data"
+              :base-type         :type/Dictionary
+              :database-type     "struct"
+              :nested-fields     #{{:name "name", :base-type :type/Text, :database-type "string", :database-position 1}}
+              :database-position 1}}
+           (#'athena/describe-table-fields-with-nested-fields :athena nested-schema))))
   (testing "sync without nested fields"
     (is (= #{{:name "id", :base-type :type/Text, :database-type "string", :database-position 0}
              {:name "ts", :base-type :type/Text, :database-type "string", :database-position 1}}
            (#'athena/describe-table-fields-without-nested-fields :athena flat-schema-columns)))))
 
+(deftest describe-table-fields-without-nested-fields-test
+  (driver/with-driver :athena
+    (is (= #{{:name "id",          :base-type :type/Integer, :database-type "integer", :database-position 0}
+             {:name "name",        :base-type :type/Text,    :database-type "string",  :database-position 1}
+             {:name "category_id", :base-type :type/Integer, :database-type "integer", :database-position 2}
+             {:name "latitude",    :base-type :type/Float,   :database-type "double",  :database-position 3}
+             {:name "longitude",   :base-type :type/Float,   :database-type "double",  :database-position 4}
+             {:name "price",       :base-type :type/Integer, :database-type "integer", :database-position 5}}
+           (jdbc/with-db-metadata [metadata (sql-jdbc.conn/db->pooled-connection-spec (mt/db))]
+             (#'athena/describe-table-fields-without-nested-fields
+              :athena
+              (#'athena/reducible-fields-from-metadata metadata nil "test_data" "venues")))))))
```

Comment:
Splitting it out into a `return-reducible-from-ResultSet` part and a `reduce-stuff` part made this easier to test